grades/management/commands/freeze_final_grades.py

```diff
@@ -22,6 +24,20 @@ def handle(self, *args, **kwargs):  # pylint: disable=unused-argument
             run = CourseRun.objects.get(edx_course_key=course_id)
         except CourseRun.DoesNotExist:
             raise CommandError('Course Run for course_id "{}" does not exist'.format(course_id))
+        try:
+            can_freeze = run.can_freeze_grades
+        except ImproperlyConfigured:
+            raise CommandError('Course Run for course_id "{}" is missing the freeze date'.format(course_id))
+        if not can_freeze:
+            raise CommandError('Course Run for course_id "{}" cannot be frozen yet'.format(course_id))
+        if CourseRunGradingStatus.is_complete(run):
+            self.stdout.write(
+                self.style.SUCCESS(
+                    'Final grades for course "{0}" are already complete'.format(course_id)
+                )
+            )
+            return
```

Comment:
i think it makes more sense to get rid of this `return` and just put the code block below into an `else` block, but that's a stylistic preference